src/Compilers/Core/Portable/Text/SourceText.cs

```diff
@@ -633,7 +633,19 @@ public virtual SourceText WithChanges(IEnumerable<TextChange> changes)
                 // there can be no overlapping changes
                 if (change.Span.Start < position)
                 {
-                    throw new ArgumentException(CodeAnalysisResources.ChangesMustBeOrderedAndNotOverlapping, nameof(changes));
+                    // Handle the case of unordered changes by sorting the input and retrying. This is inefficient, but
+                    // downstream consumers have been known to hit this case in the past and we want to avoid crashes.
+                    // https://github.com/dotnet/roslyn/pull/26339
+                    if (change.Span.End <= changeRanges.Last().Span.Start)
+                    {
+                        changes = from c in changes
+                                  where !c.Span.IsEmpty || c.NewText.Length > 0
```

Comment:
Other code in this method assumes that NewText can be null. Is the check to ensure the span isn't empty a sufficient guard against this case here?